public/app/features/explore/state/time.ts

```diff
@@ -96,13 +97,18 @@ export const updateTime = (config: {
     const range = getTimeRange(timeZone, rawRange, fiscalYearStartMonth);
     const absoluteRange: AbsoluteTimeRange = { from: range.from.valueOf(), to: range.to.valueOf() };
 
-    getTimeSrv().init(
-      new DashboardModel({
-        time: range.raw,
-        refresh: false,
-        timeZone,
-      })
-    );
+    const dashboardModel = new DashboardModel({
+      time: range.raw,
+      refresh: false,
+      timeZone,
+    });
+
+    dashboardModel.events.subscribe(TimeRangeUpdatedEvent, (event) => {
```

Comment:
I'm not sure if adding this subscription to a new instantiation with every time update will cause issues at some point. I assume if creating a new dashboardModel.events event bus every update caused us problems they would have shown themselves by now. There is a destroy() method in the dashboard model but I think we have to call it explicitly. 

Another alternative was allowing us to override the dashboard model's update time function with the `updateTimeRange` function but we would rather keep changes as contained as possible.

Reply:
This should be cleaned by garbage collector, but I agree it might be error prone.

Maybe we could do something similar you did with overrides and decouple `TimeSrv` from `DashboardModel`? At the moment we hack it to use in Explore. `TimeSrv` does not require full `DashboardModel`, just a subset of time-related properties so we could have:

```diff
export class TimeSrv {
  ...
- dashboard?: DashboardModel;
+  timeModel?: TimeModel;
}

interface TimeModel {
  fiscalYearStartMonth?: number;
  refresh: any;
  timepicker: any;
  getTimezone(): TimeZone;
  timeRangeUpdated(timeRange: TimeRange): void;
}
```

That would allow us to pass custom object, e.g. `ExploreTimeModel`, not `DashboardModel`. (it would just need to implement `getTimezone` as `timeRangeUpdated` has to be custom anyway).

In future the logic to update the URL could be decoupled as well. At the moment is dashboards-specific, but could be moved to `DashboardModel` and `ExploreTimeModel`.